### PR TITLE
HAproxy port numbers should be read from confstore.

### PR DESCRIPTION
**Pull request by shahidSTX**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered docs/notes.md](https://github.com/Seagate/cortx-s3server/blob/br/ms/ssl_cert_haproxy/docs/notes.md)
[View rendered s3backgrounddelete/readme.md](https://github.com/Seagate/cortx-s3server/blob/br/ms/ssl_cert_haproxy/s3backgrounddelete/readme.md)

### CODE REVIEW

File: scripts/provisioning/s3_haproxy_config.py [hunk 1]
```diff
@@ -59,6 +59,42 @@ def __init__(self, confstore: str):
     self.machine_id = self.provisioner_confstore.get_machine_id()
     self.logger.info(f'Machine id : {self.machine_id}')
 
+  def parse_endpoint(self, endpoint_str):
```
Comment (t7ko-seagate) [inline scripts/provisioning/s3_haproxy_config.py:62]: add `@staticmethod` before line 62.

File: scripts/provisioning/s3_haproxy_config.py [hunk 2]
```diff
@@ -59,6 +59,42 @@ def __init__(self, confstore: str):
...
+
+       Examples:
+
+       'https://s3.seagate.com:443' -> {'scheme': 'https', 'fqdn': 's3.seagate.com', 'port': '443'}
+       'https://s3.seagate.com'     -> {'scheme': 'https', 'fqdn': 's3.seagate.com'}
+       'http://s3.seagate.com:80'   -> {'scheme': 'http', 'fqdn': 's3.seagate.com', 'port': '80'}
+       'http://127.0.0.1:80'        -> {'scheme': 'http', 'fqdn': '127.0.0.1', 'port': '80'}
+    """
+    try:
+      result1 = urllib.parse.urlparse(endpoint_str)
```
Comment (t7ko-seagate) [inline scripts/provisioning/s3_haproxy_config.py:76]: add `import urllib.parse` at the top of the file.

**Review by t7ko-seagate — CHANGES_REQUESTED** [review]

File: scripts/provisioning/s3_haproxy_config.py (hunk 1 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_haproxy_config.py:66]: Addressed, please review.

File: scripts/provisioning/s3_haproxy_config.py (hunk 2 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_haproxy_config.py:80]: Done.

**Pull request merged**